Source/JavaScriptCore/dfg/DFGSpeculativeJIT.cpp

```diff
@@ -13311,6 +13311,48 @@ void SpeculativeJIT::compileStringReplace(Node* node)
     if (sample)
         m_jit.incrementSuperSamplerCount();
 
+    if (node->op() == StringReplace
+        && node->child1().useKind() == StringUse
+        && node->child2().useKind() == StringUse
+        && node->child3().useKind() == StringUse) {
+        if (JSString* replace = node->child3()->dynamicCastConstant<JSString*>(); replace && !replace->length()) {
+            SpeculateCellOperand string(this, node->child1());
+            SpeculateCellOperand search(this, node->child2());
+            GPRReg stringGPR = string.gpr();
+            GPRReg searchGPR = search.gpr();
+            speculateString(node->child1(), stringGPR);
+            speculateString(node->child2(), searchGPR);
+
+            flushRegisters();
+            GPRFlushedCallResult result(this);
+            callOperation(operationStringReplaceStringEmptyString, result.gpr(), JITCompiler::LinkableConstant(m_jit, m_graph.globalObjectFor(node->origin.semantic)), stringGPR, searchGPR);
+            m_jit.exceptionCheck();
+            cellResult(result.gpr(), node);
+            if (sample)
+                m_jit.decrementSuperSamplerCount();
```

Comment:
nit: We do this like 4 or 5 times in this function. If it were me, in the branch at the top of this function where we increment it, I'd make a function that runs on scope exit that emits the decrementing code.

Reply:
That sounds good. We have makeScopeExit for this.